blockchain/blockdao/blockdao.go

```diff
@@ -420,6 +432,12 @@ func (dao *blockDAO) PutBlock(ctx context.Context, blk *block.Block) error {
 		}
 	}
 
+	blkHash, err = dao.GetBlockHash(blkHeight)
```

Comment:
putBlock() success, this block has become the tip, so directly use blkHeight/blkHash here